src/utils/references.ts

```diff
@@ -44,3 +45,47 @@ export function stringifyReferences (paths: string[], cwd: string): string {
 export function toReference (path: string, cwd: string): string {
   return `/// <reference path="${normalizeSlashes(isAbsolute(path) ? relative(cwd, path) : normalize(path))}" />`
 }
+
+export enum DependencyTarget {
+  Main,
+  Browser
+}
+
+export enum DependencyType {
+  Ambient,
+  External
+}
+
+export interface DependencyInfo {
+  target: DependencyTarget
+  type: DependencyType
+  name: string
+}
+
+export function parseReferencePath(path: string): DependencyInfo {
+  const referencePathRegexPattern =
```

Comment:
I don't think this will work on Windows. What I'd do is wrap `parseReferences` and make them relative, then split on `path.sep`, use the first part as the "target", second as the "type", pop the last only if it's `index.d.ts` (I'd skip non-`index.d.ts` entries) and finally join the rest with `/` (because Windows will be using backslashes here.

Also, the `(\\w+)` will break on most supported values. For example, NPM scoped packages using both `/` and `@`.